api/src/components/coauthor/service.ts

```diff
@@ -158,12 +158,20 @@ export const resetCoAuthors = async (publicationId: string) => {
     const resetCoAuthors = await client.prisma.coAuthors.updateMany({
         where: {
             publicationId,
-            NOT: {
-                linkedUser: publication?.createdBy
-            }
+            OR: [
+                {
+                    NOT: {
+                        linkedUser: publication?.createdBy
+                    }
+                },
+                {
+                    linkedUser: null
+                }
+            ]
         },
         data: {
             confirmedCoAuthor: false,
+            approvalRequested: false,
```

Comment:
this change triggers the initial co-author invitation email every time the corresponding author unlocks the publication for changes and then request approvals again. 

As a co-author, if I already approved the publication, my approval should be dismissed and I should get a notification about "changes have been made...", but i'm getting the initial invitation email as well
<img width="1415" alt="image" src="https://github.com/JiscSD/octopus/assets/48569671/ea1c3f77-444a-4f95-b304-e705aedc03ee">

Reply:
That's true. Just removed this line.